dotcom-rendering/src/web/components/Nav/ExpandedMenu/Columns.tsx

```diff
@@ -116,46 +122,140 @@ const brandExtensionLink = css`
 	}
 `;
 
+const searchBar = css`
+	${from.desktop} {
+		display: none;
+	}
+	margin-left: 45px;
+	margin-bottom: 24px;
+	margin-right: 41px;
+	padding-bottom: 15px;
+`;
+
+const editionList: EditionLinkType[] = [
+	{
+		url: '/preference/edition/int',
+		id: 'INT',
+		longTitle: 'International edition',
+		title: 'International edition',
+	},
+	{
+		url: '/preference/edition/au',
+		id: 'UK',
+		longTitle: 'UK edition',
+		title: 'UK edition',
+	},
+	{
+		url: '/preference/edition/us',
+		id: 'US',
+		longTitle: 'US edition',
+		title: 'US edition',
+	},
+	{
+		url: '/preference/edition/au',
+		id: 'AU',
+		longTitle: 'Australia edition',
+		title: 'AU edition',
+	},
+];
+
+const getEdition = (editionId: EditionId): EditionLinkType => {
+	return (
+		editionList.find((edition) => edition.id === editionId) ??
+		editionList[1]
+	);
+};
+
+const getRemainingEditions = (editionId: EditionId): EditionLinkType[] => {
+	return editionList.filter((edition) => edition.id !== editionId);
+};
+
 export const Columns: React.FC<{
+	editionId: EditionId;
 	format: ArticleFormat;
 	nav: NavType;
-}> = ({ format, nav }) => (
-	<ul
-		css={columnsStyle(format.display)}
-		role="menubar"
-		data-cy="nav-menu-columns"
-	>
-		{nav.pillars.map((column, i) => (
+}> = ({ format, nav, editionId }) => {
+	const activeEdition = getEdition(editionId);
+	const remainingEditions = getRemainingEditions(activeEdition.id);
+	return (
+		<ul
+			css={columnsStyle(format.display)}
+			role="menubar"
+			data-cy="nav-menu-columns"
+		>
+			{nav.pillars.map((column, i) => (
+				<Column
+					column={column}
+					key={column.title.toLowerCase()}
+					index={i}
+					isLastColumn={i !== nav.pillars.length - 1}
+				/>
+			))}
+
+			<li>
+				<ThemeProvider theme={{ ...buttonThemeBrand }}>
+					<div css={searchBar}>
+						<LinkButton
+							href="https://www.google.co.uk/advanced_search?q=site:www.theguardian.com"
+							tabIndex={-1}
+							className="selectableMenuItem"
+							priority="secondary"
+							icon={
+								<SvgMagnifyingGlass
+									isAnnouncedByScreenReader={true}
+									size="medium"
+								/>
+							}
+							aria-label="Search with google"
+							data-link-name="nav2 : search : submit"
+							type="submit"
+						>
+							Search
+						</LinkButton>
+					</div>
+				</ThemeProvider>
+
+				<div css={lineStyle}></div>
+			</li>
+
+			<ReaderRevenueLinks readerRevenueLinks={nav.readerRevenueLinks} />
+
 			<Column
-				column={column}
-				key={column.title.toLowerCase()}
-				index={i}
+				column={{
+					...activeEdition,
+					children: remainingEditions,
+				}}
+				index={10}
+				isLastColumn={true}
```

Comment:
I think this is where we're inserting the edition dropdown?

Just a subjective thought, it might be nice to add some comment labels to this file so we can quickly see what's what. I'm not strong on this though